### PR TITLE
Added consumer infrastructure

### PR DESCRIPTION
**Pull request by HarryBosworth**
Added consumer infrastructure so that it can consume a basic message from kafka.
Added integration test to test a message can be consumed from Kafka 
Added unit tests for all supporting code.

Completes: [DSND-81](https://companieshouse.atlassian.net/browse/DSND-81) and [DSND-102](https://companieshouse.atlassian.net/browse/DSND-102)

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/officer/delta/processor/OfficerDeltaConsumerConsumerIT.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.time.Duration;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ExtendWith(MockitoExtension.class)
+class OfficerDeltaConsumerConsumerIT extends BaseKafkaIntegrationTest {
```
Comment (PhilipGlover) [inline src/test/java/uk/gov/companieshouse/officer/delta/processor/OfficerDeltaConsumerConsumerIT.java:16]: '..ConsumerConsumer' is that right?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
 management.endpoints.enabled-by-default=true
 management.endpoint.health.enabled=true
 management.endpoints.web.base-path=/officer-delta-processor
 management.endpoints.web.path-mapping.health=/healthCheck
 management.endpoint.health.show-details=ALWAYS
-
-server.port = 8080
\ No newline at end of file
+server.port=8080
+kafka.broker.url=${KAFKA_BROKER_URL:kafka:9092}
```
Comment (PhilipGlover) [inline src/main/resources/application.properties:7]: do we need a chs-configs PR to go alongside these 3 kafka properties?
also in docker-chs branch delta-add-kafdrop -  we have `KAFKA_BROKER_ADD`
and the polling items are missing

**Comment by PhilipGlover** [comment]
I've run code and seen msgs consumed.
Unit and integration tests pass (though make commands both run just the 21 unit tests)
Code looks very good apart from 2 comments above.  Ready to approve later.

File: src/test/java/uk/gov/companieshouse/officer/delta/processor/OfficerDeltaConsumerConsumerIT.java (hunk 1 above)
Comment (HarryBosworth) [reply src/test/java/uk/gov/companieshouse/officer/delta/processor/OfficerDeltaConsumerConsumerIT.java:16]: It is not. Fixed now

File: src/main/resources/application.properties (hunk 2 above)
Comment (HarryBosworth) [reply src/main/resources/application.properties:7]: Yes we do need a config PR. I was just relying on the defaults for missing vars, but being explicit, and having the variables in the properties file would be better. I'll make that change now.

Comment (HarryBosworth) [reply src/main/resources/application.properties:7]: Config PR: https://github.com/companieshouse/chs-configs/pull/1614

**Review by PhilipGlover — APPROVED** [review]
I've run code and seen msgs consumed.
Unit and integration tests pass (though make commands both run just the 21 unit tests for me)
Code looks very good and will approve corresponding docker and chs-configs.

**Pull request merged**